src/libmongoc/src/mongoc/mongoc-cursor-cmd.c

```diff
@@ -49,12 +49,17 @@ _getmore_type (mongoc_cursor_t *cursor)
    wire_version = server_stream->sd->max_wire_version;
    mongoc_server_stream_cleanup (server_stream);
 
-   if (
-      /* Server version 5.1 and newer do not support OP_GETMORE. */
-      wire_version > WIRE_VERSION_5_0 ||
-      /* Fallback to legacy OP_GETMORE wire protocol messages if exhaust cursor
-         requested. */
-      !_mongoc_cursor_get_opt_bool (cursor, MONGOC_CURSOR_EXHAUST)) {
+   const bool sharded = _mongoc_topology_get_type (cursor->client->topology) ==
+                        MONGOC_TOPOLOGY_SHARDED;
+   const bool is_exhaust =
+      _mongoc_cursor_get_opt_bool (cursor, MONGOC_CURSOR_EXHAUST);
+   const bool server_supports_getmore_no_exhaust =
+      wire_version >= WIRE_VERSION_3_6;
+   const bool server_supports_getmore_exhaust =
+      sharded ? wire_version >= WIRE_VERSION_MONGOS_EXHAUST
+              : wire_version >= WIRE_VERSION_4_2;
+   if ((is_exhaust && server_supports_getmore_exhaust) ||
+       (!is_exhaust && server_supports_getmore_no_exhaust)) {
```

Comment:
Suggest adding a comment quoting or referencing the specification corresponding to these series of checks.

Is there room to simplify these conditions per [this table](https://github.com/mongodb/specifications/blob/ddfc8b583d49aaf8c4c19fa01255afb66b36b92e/source/find_getmore_killcursors_commands.rst#exhaust)?